### PR TITLE
Kas byte counter persistence

### PR DESCRIPTION
**Pull request by alexander-kulikovskii**
closes [#1447](https://app.zenhub.com/workspaces/vision-sdk-5ca52e6c027e3e7234ac7512/issues/mapbox/mapbox-vision/1447)

### CODE REVIEW

File: MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt [hunk 1]
```diff
@@ -15,45 +17,51 @@ internal interface TotalBytesCounter {
...
 
+    fun fitInLimitCurrentSession(bytes: Long): Boolean
+
     fun fitInLimit(bytes: Long): Boolean
 
     class Impl(
         private val sessionLengthMillis: Long = SESSION_LENGTH_MILLIS,
         private val sessionMaxBytes: Long = SESSION_MAX_BYTES,
-        private val time: Time = Time.SystemImpl
+        private val time: Time = Time.SystemImpl,
```
Comment (korshaknn) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt]: maybe currentTime would be better

File: MapboxVision/src/test/java/com/mapbox/vision/sync/telemetry/TotalBytesCounterTest.kt [hunk 2]
```diff
@@ -269,23 +338,113 @@ object TotalBytesCounterTest : Spek({
...
+    val tryToSentBytes: Long,
+    val expectedValue: Boolean
+)
+
+private class TestPreference<T> : Preference<T> {
+
+    private var value: T? = null
+
+    override val preferences: SharedPreferences
+        get() = TODO("not implemented")
```
Comment (korshaknn) [inline MapboxVision/src/test/java/com/mapbox/vision/sync/telemetry/TotalBytesCounterTest.kt]: change TODO -> // do nothing

File: MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt (hunk 1 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt]: done

File: MapboxVision/src/test/java/com/mapbox/vision/sync/telemetry/TotalBytesCounterTest.kt (hunk 2 above)
Comment (alexander-kulikovskii) [reply MapboxVision/src/test/java/com/mapbox/vision/sync/telemetry/TotalBytesCounterTest.kt]: done

File: MapboxVision/src/main/java/com/mapbox/vision/utils/prefs/SessionPrefs.kt [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.mapbox.vision.mobile.core.utils.preferences.LongAdapter
+import com.mapbox.vision.mobile.core.utils.preferences.Preference
+import com.mapbox.vision.mobile.core.utils.preferences.PreferenceModel
+
+interface SessionPrefs {
+    val sessionStartMillis: Preference<Long>
+    val bytesSentPerSession: Preference<Long>
+
+    object Impl : SessionPrefs, PreferenceModel("session_prefs") {
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/utils/prefs/SessionPrefs.kt]: we already have 2 Sync modules that use 'TotalBytsCounter' and they should have different ones with different `SessionPrefs`. So in my opinion `SessionPref` -> `TotalBytesCounterStorage` and `TelemertyImpl : TotalBytesCounterStorage`

Comment (RingerJK) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/prefs/SessionPrefs.kt]: second option: have only one `TotalBytesCounterStorage` and `mpl: TotalBytesCounterStorage` but make `Impl` not `object` but just `class` with  `constructor (name: String)` and use `name` as `prefix` for _SharePreference_ strorage

Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/utils/prefs/SessionPrefs.kt]: done

File: MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt [hunk 4]
```diff
@@ -15,45 +17,51 @@ internal interface TotalBytesCounter {
...
+    fun fitInLimitCurrentSession(bytes: Long): Boolean
+
     fun fitInLimit(bytes: Long): Boolean
 
     class Impl(
         private val sessionLengthMillis: Long = SESSION_LENGTH_MILLIS,
         private val sessionMaxBytes: Long = SESSION_MAX_BYTES,
-        private val time: Time = Time.SystemImpl
+        private val currentTime: Time = Time.SystemImpl,
+        private val totalBytesCounterPrefs: TotalBytesCounterPrefs = TotalBytesCounterPrefs.Impl("")
```
Comment (RingerJK) [inline MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt]: No Need to send default value for `totalBytesCounterPrefs `

Comment (alexander-kulikovskii) [reply MapboxVision/src/main/java/com/mapbox/vision/sync/telemetry/TotalBytesCounter.kt]: done

**Pull request merged**